### PR TITLE
singular relations should always call .first to return Object / nil

### PR DESCRIPTION
**Pull request by jjmocko**

### CODE REVIEW

File: spec/teamsnap_spec.rb [hunk 1]
```diff
@@ -131,4 +131,31 @@
...
+
+    context "when a plural relation is called" do
+      it "responds with an array of objects when successful" do
+        a = event.availabilities
+        expect(a.class).to eq(Array)
+      end
+
+      it "responds with an empty array when no objects exist" do
+        a = event.assignments
+        expect(a.class).to eq(Array)
```
Comment (kyleries) [inline spec/teamsnap_spec.rb]: Do you also want to specify the size here?

**Comment by kyleries** [comment]
One suggestion, otherwise :+1:

File: spec/teamsnap_spec.rb (hunk 1 above)
Comment (jjmocko) [inline spec/teamsnap_spec.rb]: yes I do!

Comment (semmons99) [inline spec/teamsnap_spec.rb]: + expect(a.class).to eq(Array)
You can replace that with

expect(a).to be_an(Array)

On Wednesday, May 6, 2015, JJ Mocko <notifications@github.com> wrote:

> In spec/teamsnap_spec.rb
> <https://github.com/teamsnap/teamsnap_rb/pull/45#discussion_r29809469>:
>
> > @@ -131,4 +131,31 @@
> >
> >      expect(m.href).to eq("http://localhost:3000/members/1")
> >    end
> > +
> > +  context "supports relations with expected behaviors" do
> > +    let(:event) { TeamSnap::Event.find(1) }
> > +
> > +    context "when a plural relation is called" do
> > +      it "responds with an array of objects when successful" do
> > +        a = event.availabilities
> > +        expect(a.class).to eq(Array)
> > +      end
> > +
> > +      it "responds with an empty array when no objects exist" do
> > +        a = event.assignments
> > +        expect(a.class).to eq(Array)
>
> yes I do!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/teamsnap/teamsnap_rb/pull/45/files#r29809469>.
>


-- 

*Shane Emmons* | Director of Technology and Principal Architect
TeamSnap - The easiest way to manage your team
(810) 834-9542 | shane@teamsnap.com | What is TeamSnap?
<http://www.youtube.com/watch?v=fx0vRXYGijU>

File: spec/teamsnap_spec.rb [hunk 2]
```diff
@@ -131,4 +131,34 @@
...
+        a = event.assignments
+        expect(a.size).to eq(0)
+        expect(a).to be_an(Array)
+      end
+    end
+
+    context "when a singular relation is called" do
+      it "responds with the object if it exists" do
+        a = event.team
+        expect(a.class).to eq(TeamSnap::Team)
```
Comment (semmons99) [inline spec/teamsnap_spec.rb]: ditto on this being able to use `be_a` or `be_an`

**Comment by semmons99** [comment]
:shipit:

**Pull request merged**